### ISSUE TITLE
Rename `master` branch

### ISSUE THREAD

**Issue by hoechenberger** [mne-tools/mne-python#7901]
Hello,

amid the global [BLM protests](https://en.wikipedia.org/wiki/Black_Lives_Matter), it seems a number of free/open-source software projects are now [reconsidering](https://www.zdnet.com/article/github-to-replace-master-with-alternative-term-to-avoid-slavery-references/) the naming of their main source-code management branch, historically called `master` in `git`. Microsoft seems to be [actively working](https://twitter.com/natfriedman/status/1271253144442253312) on bringing out-of-the-box support for other naming schemes to GitHub.

Despite its widespread use in the contexts of software development and hardware interfaces, the term "master" **may** be considered offensive by some. While one could argue that this perception entirely disregards context, I would argue that nothing is lost by dropping the name. I would consider moving to a different name as a token of support for the struggles of ethnic minorities and marginalized people; and it would be a public display of some (granted, cheap, but still visible) efforts to actively embrace an inclusive community. Lastly I would argue that our Code of Conduct actually doesn't leave us much of a choice anyway, as it clearly [states](https://github.com/mne-tools/.github/blob/master/CODE_OF_CONDUCT.md#our-standards) we vow to use "welcoming and inclusive language". Now if the `master` term has been identified to be non-inclusive – regardless of whether **we** feel that this is true or not – it should be dropped.

Looking forward to reading your thoughts on this!

Best wishes,

–Richard

**Comment by cbrnr** [comment]
By all means change it if it makes people happy. GitHub will soon default to `main` anyway. However, I think that in this case the term master has nothing to do with master/slave (yes I know the BitKeeper background, but in a plain Git repo there is a master branch but no slave branch(es)). Same for blacklisting/whitelisting, which actually originates from the term black book (which were actual black books).

I'd only caution to wait until we know all implications of that name change. I've read that it can be non-trivial in some cases (e.g. it seems to be hard-coded for the docs), so let's do it only after GitHub releases detailed instructions how to do it.

**Comment by hoechenberger** [comment]
Hi @cbrnr 
> However, I think that in this case the term master has nothing to do with master/slave (yes I know the BitKeeper background, but in a plain Git repo there is a master branch but no slave branch(es)). Same for blacklisting/whitelisting, which actually originates from the term black book (which were actual black books).

Yes I do agree, my perception is similar. But others seem to disagree, and I don't mind making a small change to them them feel more welcome ☺️

> I'd only caution to wait until we know all implications of that name change. I've read that it can be non-trivial in some cases (e.g. it seems to be hard-coded for the docs), so let's do it only after GitHub releases detailed instructions how to do it.

Yes absolutely, just wanted to raise this question now to get a small discussion going / get an idea of what others think.

**Comment by larsoner** [comment]
Internally I don't think this change would be too difficult, but we should wait until GitHub makes it easy to redirect branch names to do this (last I looked it was not possible to redirect individual branches, only entire repos). Projects that rely on using `https://github.com/mne-tools/mne-python/zipball/master` or similar should have an opportunity to change to the new naming rather than just having things suddenly break.

**Comment by agramfort** [comment]
I am tempted to follow here. If core projects that are models for us do
this then we could follow.

**Comment by GuillaumeFavelier** [comment]
Not directly related but I should probably rename the `time_viewer` selected as reference in `LinkViewer`:

https://github.com/mne-tools/mne-python/blob/master/mne/viz/_brain/_timeviewer.py#L1221-L1222

**Comment by hoechenberger** [comment]
> Not directly related but I should probably rename the `time_viewer` selected as reference in `LinkViewer`:
> 
> https://github.com/mne-tools/mne-python/blob/master/mne/viz/_brain/_timeviewer.py#L1221-L1222

Heh. Do we have a good alternative? `main`, perhaps?

**Comment by GuillaumeFavelier** [comment]
`main` or just simply `reference` should do the trick?

**Comment by hoechenberger** [comment]
> `main` or just simply `reference` should do the trick?

How about
```python
        controller = self.time_viewers[0]  # select the first time viewer, which controls the scene
        value = controller.time_call.slider_rep.GetValue()
```

**Comment by cbrnr** [comment]
Feel free to rename, but master without slaves should be fine (because masters can also have apprentices).

**Comment by GuillaumeFavelier** [comment]
Thanks all for the suggestions, I decided to rename into `leader` because it suits the behaviour perfectly. And sorry for the noise on the issue.

**Comment by hoechenberger** [comment]
The Linux kernel developers just [decided](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=49decddd39e5f6132ccd7d9fdc3d7c470b0061bb) the use the following terms:

```
+For symbol names and documentation, avoid introducing new usage of
+'master / slave' (or 'slave' independent of 'master') and 'blacklist /
+whitelist'.
+
+Recommended replacements for 'master / slave' are:
+    '{primary,main} / {secondary,replica,subordinate}'
+    '{initiator,requester} / {target,responder}'
+    '{controller,host} / {device,worker,proxy}'
+    'leader / follower'
+    'director / performer'
+
+Recommended replacements for 'blacklist/whitelist' are:
+    'denylist / allowlist'
+    'blocklist / passlist'
+
+Exceptions for introducing new usage is to maintain a userspace ABI/API,
+or when updating code for an existing (as of 2020) hardware or protocol
+specification that mandates those terms. For new specifications
+translate specification usage of the terminology to the kernel coding
+standard where possible.
```

**Comment by cbrnr** [comment]
See https://github.com/github/renaming/

**Comment by hoechenberger** [comment]
Just for reference, the VS Code Python Extension is chiming in now, too:
https://github.com/microsoft/vscode-python/issues/13645

**Comment by cbrnr** [comment]
Should we go for it? https://docs.github.com/en/github/administering-a-repository/renaming-a-branch

**Comment by hoechenberger** [comment]
> Should we go for it? https://docs.github.com/en/github/administering-a-repository/renaming-a-branch

I just came here to ask the same, do: yes, let's do that!!

**Comment by agramfort** [comment]
have you done with any other project ? with one with many forks like mne?

Let's not experiment with the mne-python repo

>

**Comment by cbrnr** [comment]
What about starting with `mne-testing-data`? I've prepared all changes for this repo in #8758.

**Comment by agramfort** [comment]
I'll just speak for myself. I have tons of stuff to do and I need to
prioritize things.
Doing things like this is very low priority right now.

So if you want to do it fine. But I will try **not** to look and let you in
charge.

>

**Comment by cbrnr** [comment]
Same, this is also low priority for me. I don't want to break anything, so I'd rather not do it unless others look very carefully at the changes.

**Comment by hoechenberger** [comment]
The GH docs specifically say that all existing references to the old branch names will continue to work:

```
If you rename a branch, GitHub will automatically redirect links on GitHub.com that contain the old branch name to the equivalent link on the renamed branch. GitHub will also update branch protection policies, as well as the base branch for open pull requests and draft releases.
```

**Comment by cbrnr** [comment]
Yes, it should work without #8758. But what if it doesn't? Are there any larger projects that have already switched? I checked numpy, matplotlib, pandas, scipy, scikit-learn - they are all still using master.

**Comment by hoechenberger** [comment]
> Yes, it should work without #8758. But what if it doesn't? Are there any larger projects that have already switched? I checked numpy, matplotlib, pandas, scipy, scikit-learn - they are all still using master.

If it doesn't, we just rename it back. I don't understand the hesitation honestly…

**Comment by hoechenberger** [comment]
> Are there any larger projects that have already switched? I

vscode-python has switched: https://github.com/microsoft/vscode-python/issues/13645

**Comment by hoechenberger** [comment]
I just renamed `master` to `main` on https://github.com/hoechenberger/questplus

After that I had to run locally:
```
git branch -m master main
git fetch origin
git branch -u origin/main main
```
All working now.

**Comment by cbrnr** [comment]
@hoechenberger and I will rename the branch this evening.

**Comment by drammock** [comment]
I'd like to request that you do this next weekend instead of today, and that you announce it on the mailing list and forum first. Advance warning may avoid a lot of surprises and complaints.

**Comment by cbrnr** [comment]
OK, no problem!

**Comment by cbrnr** [comment]
I just renamed the master branch to main in MNELAB, everything is fine (also locally).

**Comment by cbrnr** [comment]
Done - see https://mne.discourse.group/t/we-have-renamed-the-main-development-branch-on-github-to-main/2526!
